lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -1,1434 +1,70 @@
-        [ "Did fetch values "
-        , fmt $ mapF m
-        ]
-    toText (MsgDidChange m) = mconcat
-        [ "New values: "
-        , fmt $ mapF m
-        ]
-    toText (MsgAddedObserver key) = mconcat
-        [ "Started observing values for key "
-        , pretty key
-        ]
-    toText (MsgRemovedObserver key) = mconcat
-        [ "Stopped observing values for key "
-        , pretty key
-        ]
-
-{-------------------------------------------------------------------------------
-    Local State Query Helpers
--------------------------------------------------------------------------------}
-
-byronOrShelleyBased
-    :: LSQ Byron.ByronBlock m a
-    -> (forall shelleyEra. LSQ (Shelley.ShelleyBlock (shelleyEra StandardCrypto)) m a)
-    -> LSQ (CardanoBlock StandardCrypto) m a
-byronOrShelleyBased onByron onShelleyBased = onAnyEra
-    onByron
-    onShelleyBased
-    onShelleyBased
-    onShelleyBased
-    onShelleyBased
-
--- | Create a local state query specific to the each era.
---
--- This combinator treats @MismatchEraInfo@ as impossible, which is true if the
--- @LSQEra@ value the @LSQ@ interpreter uses always matches the era of the
--- acquired point.
---
--- Where possible, the more convenient @shelleyBased@ or @byronOrShelleyBased@
--- should be used. This more raw helper was added to simplify dealing with
--- @PParams@ in alonzo.
-onAnyEra
-    :: LSQ Byron.ByronBlock m a
-    -> LSQ (Shelley.ShelleyBlock StandardShelley) m a
-    -> LSQ (Shelley.ShelleyBlock StandardAllegra) m a
-    -> LSQ (Shelley.ShelleyBlock StandardMary) m a
-    -> LSQ (Shelley.ShelleyBlock StandardAlonzo) m a
-    -> LSQ (CardanoBlock StandardCrypto) m a
-onAnyEra onByron onShelley onAllegra onMary onAlonzo = currentEra >>= \case
-    AnyCardanoEra ByronEra -> mapQuery QueryIfCurrentByron onByron
-    AnyCardanoEra ShelleyEra -> mapQuery QueryIfCurrentShelley onShelley
-    AnyCardanoEra AllegraEra -> mapQuery QueryIfCurrentAllegra onAllegra
-    AnyCardanoEra MaryEra -> mapQuery QueryIfCurrentMary onMary
-    AnyCardanoEra AlonzoEra -> mapQuery QueryIfCurrentAlonzo onAlonzo
-  where
-    mapQuery
-        :: (forall r. BlockQuery block1 r
-            -> BlockQuery block2
-                ((Either (MismatchEraInfo (CardanoEras StandardCrypto))) r))
-        -> LSQ block1 m a
-        -> LSQ block2 m a
-    mapQuery _ (LSQPure x) = LSQPure x
-    mapQuery f (LSQBind ma f') = LSQBind (mapQuery f ma) (mapQuery f . f')
-    mapQuery f (LSQry q) = unwrap <$> LSQry (f q)
-
-    unwrap = either (error "impossible: byronOrShelleyBased query resulted in an \
-        \era mismatch") id
+        NodeNetworkLog l -> getSeverityAnnotation l
+        BlockfrostNetworkLog l -> getSeverityAnnotation l
 
+withNetworkLayer
+    :: HasCallStack
+    => Tracer IO NetworkLayerLog
+    -> BlockchainSource
+    -> Cardano.NetworkId
+    -> NetworkParameters
+    -> SyncTolerance
+    -> ContT r IO (NetworkLayer IO (CardanoBlock StandardCrypto) )
```

Comment:
Hm, it may be difficult for the blockfrost backend to return a full `CardanoBlock StandardCrypto` as opposed to our own, simplified `Block` type. 🤔 Hence, it could be sensible to insert the conversion here; consumers who want a full `CardanoBlock StandardCrypto` can always call `Node.withNetworkLayer` directly.

Can be postponed to later. I think the only consumer who needs the full block is the stake pool layer, but that one will have to be adapated to light-mode anyway.